src/runtime/loader.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+
+namespace Python.Runtime
+{
+    using static Runtime;
+
+    [Obsolete("Only to be used from within Python")]
+    static class Loader
+    {
+        public static int Initialize(IntPtr data, int size)
```

Comment:
I'm not sure what you mean. The `clr_loader` package by itself just allows us to initialize a runtime, load `Python.Runtime.dll` and call a function with this signature (`(IntPtr, Int32) -> Int32`). Everything else is in the `pythonnet` repository. I don't plan allowing different versions of `Python.Runtime.dll` to be used with the `clr.py` or `pythonnet` package that we distribute.

Reply:
Can you provide an example?

I am talking about a scenario when we need to change the value from simply `string` DLL name to some data structure with more than 1 field (imagine we have to also pass Python version here in the future).

Reply:
I have no idea what kind of example you expect :). Both the caller and the function are part of the pythonnet codebase. The signature will always be what it is now as that is the only thing that is reasonably supported. We have to serialise what we want to transmit in some way, for now it is enough to use UTF8 on the string. If we want to change what is sent here, we do it on both sides.

Reply:
Doesn't one of the sides end up in a separate pip package (clr loader)? So if I install clr loader using pip, then later install pythonnet using pip, pythonnet then must have exactly same major version of clr loader as the dependency, so if clr loader is too old, it would be updated.

Reply:
No. `clr-loader` is very much a completely stripped down version of `pythonnet`: It allows one to load one of the supported runtimes (Mono, .NET Framework, .NET Core) and call a function that has (exactly) the signature `int func(IntPtr, int size)`. On the `pythonnet` side we use that from within Python (cf. `pythonnet/__init__.py:load`) to load the `Python.Runtime.dll` that came with the installation:

```python
    _LOADER_ASSEMBLY = _RUNTIME.get_assembly(dll_path)

    func = _LOADER_ASSEMBLY["Python.Runtime.Loader.Initialize"]
    if func(f"{libpython or ''}".encode("utf8")) != 0:
        raise RuntimeError("Failed to initialize Python.Runtime.dll")
```